tests/std/tests/P0896R4_ranges_alg_generate/test.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::generate, ranges::iterator_t;
+    using std::same_as;
+
+    {
+        int output[]    = {13, 42, 1367};
+        const int value = 7;
+        auto result     = generate(std::begin(output), std::end(output), []() { return value; });
+        for (auto elem : output) {

```

Comment:
As mentioned in https://github.com/microsoft/STL/pull/904#discussion_r443809863 :

```suggestion
        for (const auto& elem : output) {
```